src/batch/src/executor/iceberg_scan.rs

```diff
@@ -214,6 +234,13 @@ impl BoxedExecutorBuilder for IcebergScanExecutorBuilder {
         let schema = Schema::new(fields);
         let metrics = source.context.batch_metrics().clone();
 
+        if split_list.is_empty() {
+            return Ok(Box::new(IcebergScanNoDataExecutor {
+                schema,
+                identity: source.plan_node().get_identity().clone(),
+            }));
+        }
```

Comment:
It seems `IcebergScanNoDataExecutor` is unnecessary, if `split_list.len = 1` is constantly true.

Reply:
Fixed.